kubernetes/kubernetes.go

```diff
@@ -177,14 +182,16 @@ func newJobSpec(config KubernetesRunnerConfig, rsrcReqs v1.ResourceRequirements)
 }
 
 type KubernetesRunnerConfig struct {
-	EnvVars  map[string]string
-	Resource metadata.ResourceID
-	Image    string
-	NumTasks int32
-	Specs    metadata.KubernetesResourceSpecs
+	EnvVars   map[string]string
+	JobPrefix string
+	Resource  metadata.ResourceID
+	Image     string
+	NumTasks  int32
+	Specs     metadata.KubernetesResourceSpecs
 }
 
 type JobClient interface {
+	GetJobName() string
```

Comment:
I added this for logging. 

I actually think our design for this could be better. It feels wrong that a job client stores the job name information. It'd be like creating an HttpClient interface that has the URL hardcoded to it. Doesn't make sense to me.

My suggestion would be to have a lean job client class and have the runners contain Job information. Or make have a Job type



```
type JobClient interface

Get(job Job)
GetCronJob(job Job)
Watch(job Job)


```